clients/python/lakefs_sdk/client.py

```diff
@@ -22,20 +22,37 @@
 class _WrappedApiClient(ApiClient):
     """ApiClient that fixes some weirdness"""
 
-    # Wrap files_parameters to work with unnamed "files" (e.g. MemIOs).
     def files_parameters(self, files=None):
-        if files is not None:
-            for (param_name, file_instances) in files.items():
-                i = 0
-                if file_instances is None:
-                    continue
-                for file_instance in file_instances:
-                    if file_instance is not None and not hasattr(file_instance, 'name'):
-                        # Generate a fake name.
-                        i += 1
-                        file_instance.name = f'{param_name}{i}'
-        return super().files_parameters(files)
+        """
+        Transforms input file data into a formatted list and combines it with superclass file parameters.
+        In case file_name is a string we assume that it is a path to the file that needs to be read.
+        In case file_name is bytes we assume that it is a file-like object that we append the information.
+        The parent class will handle the files to read.
+        """
+        if not files:
+            return []
 
+        params = []
+        files_to_read = {}
+        item_index = 0
+
+        for key, value in files.items():
```

Comment:
You could save a line below by doing this:

```suggestion
        for idx, (key, value) in enumerate(files.items()):
```

And then set `name = f'{key}{idx + 1}`, if you'd like.

Related: Does this mean that files uploaded by name via SDK will never have a mimetype set? I noticed this when ingesting a `.py` file lately, so this seems to be the case currently in the client, too?

Reply:
Thanks I'll use it - idx can start with 0 too, just need to have something to identify the different content.

Related to the SDK content type - true. The SDK generated code auto detect content type by using filename.
As limited currently by the type of parameter this method accepts as `content`, did see a quick workaround for passing content type or filename.

This PR will only open the option of upload content which is not based on local file.